File: r/sparklabs/batch/sales/dataframe/orders_per_city.R
source("r/sparklabs/batch/sales/dataframe/utils.R")

joined <- join(orders, customers, orders$customer_id == customers$customer_id)
result <- orderBy(
  agg(groupBy(joined, joined$city), total_orders_per_city = count(joined$customer_id)),
  desc("total_orders_per_city")
)
showDF(result, truncate = FALSE)

File: scala/src/main/scala/io/devorbit/sparklabs/batch/sales/dataframe/OrdersPerCity.scala
package io.devorbit.sparklabs.batch.sales.dataframe

import org.apache.spark.sql.functions._

object OrdersPerCity extends App {

  Utils.orders.join(Utils.customers, Seq("customer_id"), "inner")
    .groupBy(col("city"))
    .agg(count("*").as("total_orders_per_city"))
    .orderBy(desc("total_orders_per_city"))
    .show(false)

//  Thread.sleep(1000000)
}
